### PR TITLE
generic-worker revision 3

### PR DESCRIPTION
**Pull request by aerickson**
### changes from revision 2 (https://github.com/bclary/mozilla-bitbar-docker/pull/4):
  - Adds the script.py output back into the taskcluster live.log file.

This change requires payload modifications (see https://phabricator.services.mozilla.com/D23918 for an idea of the changes required - that change still has testing code/not ready to land.)

### attempt 1

Store the original payload command and have generic worker run script.py, that will then call the original payload. Didn't work because g-w won't allow any extra fields in the payload.

docker image label: `20190320T152017`

### attempt 2

Make script.py take all args passed in as the command to run.

build 1: `20190320T174624`: bad,  payload not sent in on stdin.
build 2: `20190320T181216`: bad, didn't account for the '--'
build 3: `20190320T191220`: bad, command wasn't quoted.
build 4: `20190320T195157`: bad, don't want login shell.
build 5 (44826c1 ): `20190320T200537`: success (test runs: https://treeherder.mozilla.org/#/jobs?repo=try&revision=6dc2d8a9d3682792d44a2d116c3755a341dfd00f)
build 6 (c139f75): `20190322T150747`: success (test runs: https://treeherder.mozilla.org/#/jobs?repo=try&revision=ef102c0b2e932c03c2f52bb2a40721380f450905)
build 7 (da018f8): `20190325T102054`: seemed to work (test runs: https://treeherder.mozilla.org/#/jobs?repo=try&revision=124da02234e5ba31f98275bc50d9d576892ed4b6)
build 8 (e933a29): `20190325T160701`: seems good (test runs: https://treeherder.mozilla.org/#/jobs?repo=try&revision=f4703ccceac2864d3095537e496a35b4ecc9b95b)
build 9(unpushed, contains proposed fix for adb key issue on batt devices): `20190408T152200`: worked on batt devices, but on non-batt got error: `ln: '/root/.android/adbkey' and '/builds/worker/.android/adbkey' are the same file`
build 10(b74d06b): `20190408T165211`

### CODE REVIEW

File: scripts/entrypoint.py
Comment (bclary) [inline scripts/entrypoint.py]: shall we just go ahead and remove these comments?

File: Dockerfile
Comment (bclary) [inline Dockerfile:71]: Do you think it is worth it to go ahead and install https://github.com/taskcluster/livelog ?

File: taskcluster/script.py
Comment (bclary) [inline taskcluster/script.py]: What do you think of using argparse and https://docs.python.org/2.7/library/argparse.html#argparse.ArgumentParser.parse_known_args rather than -- ? Is there any benefit?

script.py could define --command to pass test_command and the args remaining after parse_known_args could be passed to it. This how I would have approached it but apart from "feelings" I can't really claim it would be "better". I'll leave it up to you.

File: scripts/entrypoint.py
Comment (aerickson) [reply scripts/entrypoint.py]: Yeah, will do.

File: Dockerfile
Comment (aerickson) [reply Dockerfile:71]: Yeah, I did it in another branch (https://github.com/bclary/mozilla-bitbar-docker/compare/master...aerickson:gw_v2_1), will port it over.

File: taskcluster/script.py
Comment (aerickson) [reply taskcluster/script.py]: Aaah, I didn't know about that. Yeah, sounds good. Gets me out of the testing business. :)

File: taskcluster/script.py
Comment (bclary) [inline taskcluster/script.py]: I'd rather not have these in global scope. Let's move the parser stuff into main and just do it there. Everything else looks great.

Comment (aerickson) [reply taskcluster/script.py]: Changed.

**Review by bclary — APPROVED** [review]
r+. lgtm. hold off merging till we're ready with the other things.

**Pull request merged**
